docs/tensorflow.md

```diff
@@ -1,135 +1,238 @@
 
-For examples of code structure to apply the hook wrappers, see the [Examples](#examples) section.
+For examples of code structures that you can use to apply the hook wrappers, see the [Code Examples](#examples) section.
 
-#### 4. Take actions using the hook APIs
+### Step 3: Register the hook to model.fit() <a name="register-a-hook"></a>
 
-For a full list of actions that the hook APIs offer to construct hooks and save tensors, see [Common hook API](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md#common-hook-api) and [TensorFlow specific hook API](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md#tensorflow-specific-hook-api).
+To collect the tensors from the hooks that you registered, add `callbacks=[hook]` to the Keras `model.fit()` API. This will pass the SageMaker Debugger hook as a Keras callback. Similarly, add `hooks=[hook]` to the `MonitoredSession()`, `tf.function()`, and `tf.estimator()` APIs. For example:
 
->**Note**: The `inputs`, `outputs`, and `layers` collections are not currently available for TensorFlow 2.1.
+```python
+model.fit(X_train, Y_train,
+          batch_size=batch_size,
+          epochs=epoch,
+          validation_data=(X_valid, Y_valid),
+          shuffle=True,
+          # smdebug modification: Pass the hook as a Keras callback
+          callbacks=[hook])
+```
+
+### Step 4: Perform actions using the hook APIs
+
+For a full list of actions that the hook APIs offer to construct hooks and save tensors, see [Common hook API](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md#common-hook-api) and [TensorFlow specific hook API](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md#tensorflow-specific-hook-api).
 
 ---
 
-## Examples
+## Code Examples <a name="examples"></a>
 
-The following examples show the three different hook constructions of TensorFlow. The following examples show what minimal changes have to be made to enable SageMaker Debugger while using the AWS containers with script mode. To learn how to use the high-level Debugger features with zero script change on AWS Deep Learning Containers, see [Use Debugger in AWS Containers](https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-container.html).
+The following code examples show the base structures that you can use for hook registration in various TensorFlow training scripts. If you want to use the high-level Debugger features with zero script change on AWS Deep Learning Containers, see [Use Debugger in AWS Containers](https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-container.html).
 
 ### Keras API (tf.keras)
 ```python
 import smdebug.tensorflow as smd
 
-hook = smd.KerasHook(out_dir=args.out_dir)
+hook = smd.KerasHook.create_from_json_file()
 
 model = tf.keras.models.Sequential([ ... ])
 model.compile(
     optimizer='adam',
     loss='sparse_categorical_crossentropy',
 )
 # Add the hook as a callback
+hook.set_mode(mode=smd.modes.TRAIN)
```

Comment:
comment here, hook.set_mode will help you differentiate between tensors in TRAIN and EVAL phases. DEtails about mode here[link]